developer_manual/general/security.rst

```diff
@@ -229,6 +229,34 @@ To prevent CSRF in an app, be sure to call the following method at the top of al
 
 If you are using the App Framework, every controller method is automatically checked for CSRF unless you explicitly exclude it by setting the @NoCSRFRequired annotation before the controller method, see :doc:`../app/controllers`
 
+When using forms in your app or theme, you need to supply a csrf-token to be checked when submitting the form. Insert a token like this:
+
+.. code-block:: php
+  
+  // in itemController.php
+  ...
+  function renderItemTemplate() {
+    ...
+    return new TemplateResponse('appname', 'itemTemplate', array(
+      'items' => ...,
+      'requesttoken' => (\OC::$server->getSession()) ? \OCP\Util::callRegister() : ''
```

Comment:
What's the `->getSession()` check used for? Wouldn't it be easier and sufficient to just use a local variable in the template?

Something like
```php
<?php $CSRFToken = \OCP\Util::callRegister(); ?>
… value="<?php p($CSRFToken); ?>" …
```

Reply:
Yeah, maybe you're right. I mean, it would be best practice to get it from the controller, right? But for a short documentation example your snippet is probably sufficient.
As for the `\OC::$server->getSession()`, I have no idea what it does, because I can't find it in the documentation 😆 I've incorporated that because I've seen apps do it like this. Maybe it's wrong? Can you say something about it?